node/block_propagate_test.go

```diff
@@ -30,18 +30,42 @@ func connect(t *testing.T, nd1, nd2 *Node) {
 	}
 }
 
+func requireMineOnce(ctx context.Context, t *testing.T, minerNode *Node) *types.Block {
+	head := minerNode.ChainReader.GetHead()
+	headTipSet, err := minerNode.ChainReader.GetTipSet(head)
+	require.NoError(t, err)
+	baseTS := headTipSet
+	require.NotNil(t, baseTS)
+
+	worker, err := minerNode.CreateMiningWorker(ctx)
+	require.NoError(t, err)
+
+	// Miner should win first election as it has all the power so only
+	// mine once with 0 null blocks
+	out := make(chan mining.Output)
+	var wonElection bool
+	var wg sync.WaitGroup
+	wg.Add(1)
+	go func() {
+		wonElection = worker.Mine(ctx, headTipSet, 0, out)
+		wg.Done()
+	}()
+	next := <-out
```

Comment:
Why the goroutine and channel suff here?

Reply:
Just to interact with the miner worker directly.  I didn't want to call `StartMining` which I think I'd need to do to properly use the block api's `MineOnce` function.